target/linux/ramips/mt7621/base-files/etc/board.d/03_gpio_switches

```diff
@@ -10,6 +10,9 @@ case "$board" in
 mikrotik,routerboard-750gr3)
 	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "17"
 	;;
+mikrotik,routerboard-760igs)
+	ucidef_add_gpio_switch "poe_passthrough" "PoE Passthrough" "497"
+	;;
```

Comment:
I need some help here:

For 750gr3
Manufacturer page states 750gr3 has PoE in, but no PoE out (760igs has PoE out port 5).
The commit which added this for 750gr3: https://github.com/openwrt/openwrt/commit/6ba58b7b020c4b793e6fb7a43c8adef070579d8d states GPIO 7 is PoE input status.
Is the 750gr3 line for GPIO7 (PoE in? status) + base number, and should it be an input, not a switch (which configures GPIO direction as out)?
Or is it GPIO17, for a poe_passthrough that I cannot see mentioned, or understand for 750gr3?

@llogar Could you please clarify if GPIO17 means something to you, or it is just reflecting the 750gr3 lines above? Thanks

I am not sure how to check which GPIO number is correct?

I do have a Mikrotik passive PoE injector. I could try that later, export the unused GPIOs as input, and check which GPIO goes high?

I see this on device:
```
root@OpenWrt:~# dmesg | grep -i gpio
[    0.519532] mt7621_gpio 1e000600.gpio: registering 32 gpios
root@OpenWrt:~# ls /sys/class/gpio/gpiochip4
gpiochip416/  gpiochip448/  gpiochip480/
root@OpenWrt:~# cat /sys/class/gpio/gpiochip4*/base
416
448
480
```

With SFP module working:
```
root@OpenWrt:~# cat /sys/kernel/debug/gpio 
gpiochip2: GPIOs 416-447, parent: platform/1e000600.gpio, 1e000600.gpio-bank2:

gpiochip1: GPIOs 448-479, parent: platform/1e000600.gpio, 1e000600.gpio-bank1:

gpiochip0: GPIOs 480-511, parent: platform/1e000600.gpio, 1e000600.gpio-bank0:
 gpio-486 (                    |los                 ) in  lo IRQ 
 gpio-487 (                    |sysfs               ) in  lo 
 gpio-488 (                    |mod-def0            ) in  lo IRQ ACTIVE LOW
 gpio-489 (                    |routerboard-760igs:b) out hi 
 gpio-490 (                    |tx-disable          ) out lo 
 gpio-492 (                    |usb_power           ) out hi 
 gpio-493 (                    |mode                ) in  hi IRQ 
 gpio-495 (                    |beeper              ) out lo 
 gpio-496 (                    |routerboard-760igs:b) out hi 
 gpio-497 (                    |sysfs               ) out lo 
 gpio-498 (                    |reset               ) in  hi IRQ 
```

The best discussion of GPIOs I can find seems to be https://github.com/openwrt/openwrt/pull/1814#issuecomment-547520975

The device also supports PoE out on ether5. If I wanted to investigate that, is the best way to return to OEM firmware, and have a poke around?

Reply:
I am pretty sure, GPIO7 (gpio-487) is poe-status LED and GPIO17 (gpio-497) is poe-passthrough, I don't have the device here with me, but I can definitly confirm this on Monday...

Reply:
Great, thank you!

Could you please explain for me what poe_passthrough does physically?
I am just not following what it would switch on 750gr3.

Could you also point me in the direction of how you work out which GPIOs are connected to what? I source the LEDs in the OEM GPL source, but that was all.

Cheers!

Reply:
PoE passthrough is, as the name suggests, a passive passthrough (to the last switch port) of the power applied to the PoE in port.

If @llogar can give us the output of `cat /sys/firmware/mikrotik/hard_config/hw_options` on the 750GR3, that would help. Vendor code shows possible support for PoE passthrough on the 750GR3, conditional on 'has POE out'.

For reference, here's the relevant platform data:
```
static struct platform_device ralink_poeout_device = {
        .name   = "poe-out-simple",
        .id     = -1,
        .dev    = {
                .platform_data = &(struct poeout_port){
                       .eth_port = 4,
                       .gpo_on = 17,
                       .gpi_status = 7,
               },
        },
};
```

Reply:
Thank you very much! I just didn't see PoE out mentioned on the Mikrotik 750gr3 overview page, and wanted to make sense of it to understand if it applied to 760igs

Should the 750gr3 setting have a GPIO base added to it then? I do not have the device to check.

the 760igs hw_options looks fine:
```
root@OpenWrt:~# cat /sys/firmware/mikrotik/hard_config/hw_options 
raw		: 0x08534006

no UART		: false
has Vreg	: true
has usb		: true
has ATtiny	: false
no NAND		: true
has LCD		: false
has POE out	: true
has MicroSD	: true
has SIM		: false
has SFP		: true
has WiFi	: false
has TS ADC	: true
has PLC		: false
```

Reply:
I'm not sure how that gpio base is handled nowadays. What's certain is that the gpio line is the same for both devices.

Reply:
Would like to find someone with a 750gr3 then…
to check:
```
cat /sys/firmware/mikrotik/hard_config/hw_options
cat /sys/kernel/debug/gpio
logread | grep -i gpio
```

Hi @pmelange, I saw you mentioned running a 750gr3 in a recent DTS improvements PR. If you get a chance, could you please show us the output of these command on the 750gr3? Thanks!

Reply:
```
uname -a

Linux scherer8 5.4.41 #0 SMP Fri Sep 6 08:29:50 2019 mips GNU/Linux
```

```
cat /sys/firmware/mikrotik/hard_config/hw_options 
raw		: 0x084a4007

no UART		: true
has Vreg	: true
has usb		: true
has ATtiny	: false
no NAND		: true
has LCD		: false
has POE out	: false
has MicroSD	: true
has SIM		: false
has SFP		: false
has WiFi	: false
has TS ADC	: true
has PLC		: false
```

```
cat /sys/kernel/debug/gpio
gpiochip2: GPIOs 416-447, parent: platform/1e000600.gpio, 1e000600.gpio-bank2:

gpiochip1: GPIOs 448-479, parent: platform/1e000600.gpio, 1e000600.gpio-bank1:

gpiochip0: GPIOs 480-511, parent: platform/1e000600.gpio, 1e000600.gpio-bank0:
 gpio-480 (                    |routerboard-750gr3:g) out hi 
 gpio-492 (                    |usb_power           ) out hi 
 gpio-493 (                    |mode                ) in  hi IRQ 
 gpio-495 (                    |beeper              ) out lo 
 gpio-496 (                    |routerboard-750gr3:b) out hi 
 gpio-498 (                    |reset               ) in  hi IRQ 
```

My logbuffer is way past the booting stage (uptime is currentl 26 days) so I grep'ed dmesg instead.
```
dmesg | grep -i gpio
[    0.522795] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.528614] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.534431] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.824537] gpio-export gpio_export: 1 gpio(s) exported
[   31.569345] export_store: invalid GPIO 17
```

Reply:
`has POE out	: false`
That answers it. No POE out on this board.

Reply:
Thank you very much!

Okay, with 750gr3, given:
> has POE out	: false
> [   31.569345] export_store: invalid GPIO 17

Should the 750gr3 03_gpio_switches poe_passthrough be removed, or changed to 497 and shared with 760igs?

Reply:
I think it should be removed since it's not enabled on the hardware.

Reply:
I agree that it should be removed.  having the ```export_store: invalid GPIO 17``` message in the logfile and the automatically generated part in /etc/config/system could confuse people.

Reply:
PR with this change, and one (trivial) DTS change here: https://github.com/openwrt/openwrt/pull/3107